awd/helpers.py

```diff
@@ -174,29 +232,120 @@ def create_dss_message_body(
+            and self.valid_result_message_body(sqs_message=sqs_message)
+            and sqs_message.get("ReceiptHandle")
+        ):
+            doi = sqs_message["MessageAttributes"]["PackageID"]["StringValue"]
+            doi_process_attempt = DoiProcessAttempt.get(doi)
+            message_body = json.loads(str(sqs_message["Body"]))
+            receipt_handle = sqs_message["ReceiptHandle"]
+            if message_body["ResultType"] == "error":
+                self.error_message(
+                    doi=doi,
+                    message_body=message_body,
+                    receipt_handle=receipt_handle,
+                    doi_process_attempt=doi_process_attempt,
+                    retry_threshold=retry_threshold,
+                )
+            else:
+                self.success_message(
+                    doi=doi,
+                    message_body=message_body,
+                    receipt_handle=receipt_handle,
+                    doi_process_attempt=doi_process_attempt,
+                )
+        else:
+            raise InvalidSQSMessageError
+
+    def error_message(
+        self,
+        doi: str,
+        message_body: str,
+        receipt_handle: str,
+        doi_process_attempt: DoiProcessAttempt,
+        retry_threshold: str,
+    ) -> None:
+        """Process an error result message.
+
+        Args:
+            doi: The DOI of the article.
+            message_body: The body of the SQS result message.
+            receipt_handle: The receipt handle of the SQS result message.
+            doi_process_attempt: A database item represented as a PynamoDB object.
+            retry_threshold: The number of times to attempt processing an article.
+        """
+        logger.exception("DOI: %s, Result: %s", doi, message_body)
+        self.delete(receipt_handle)
+        if doi_process_attempt.attempts_exceeded(retry_threshold=int(retry_threshold)):
+            doi_process_attempt.update_status(status_code=Status.FAILED.value)
+            logger.exception(
+                "DOI: '%s' has exceeded the retry threshold and will not be "
+                "attempted again.",
+                doi,
+            )
+        else:
+            doi_process_attempt.update_status(status_code=Status.UNPROCESSED.value)
+
+    def success_message(
+        self,
+        doi: str,
+        message_body: str,
+        receipt_handle: str,
+        doi_process_attempt: DoiProcessAttempt,
+    ) -> None:
+        """Process an success result message.
+
+        Args:
+            doi: The DOI of the Article.
+            message_body: The body of the SQS result message.
+            receipt_handle: The receipt handle of the SQS result message.
+            doi_process_attempt: A database item represented as a PynamoDB object.
+        """
+        logger.info("DOI: %s, Result: %s", doi, message_body)
+        self.delete(receipt_handle)
+        doi_process_attempt.update_status(status_code=Status.SUCCESS.value)
```

Comment:
I hate to say it, but I might suggest moving these as methods on the `DoiProcessAttempt` class.  

Other than logging and deleting the SQS `receipt_handle`, it looks like they are mostly performing work on the Dynamodb table row?  updating the status based on success or error of SQS message?

Seems reasonable for this client to 

1. extract the DOI from the message (very much in SQS territory)
2. and then get that row / model instance via `doi_process_attempt = DoiProcessAttempt.get(doi)`
3. parse and determine from SQS message if the work was successful

But then, it feels like the responpsibility of DynamoDB work falls to `DoiProcessAttempt` more than this client, which ideally might just be to read / parse / delete SQS messages.

I could see the calling flow from `process_result_message()` something like this:

```python
# prior work in SQSClient.process_result_message()...

# get DOI from message and retrieve row from database
doi = sqs_message["MessageAttributes"]["PackageID"]["StringValue"]
doi_process_attempt = DoiProcessAttempt.get(doi)

# parse SQS message
message_body = json.loads(str(sqs_message["Body"]))
receipt_handle = sqs_message["ReceiptHandle"]
self.delete(receipt_handle) # look like never used, and both success/error delete it

if message_body["ResultType"] == "error":
    logger.exception("DOI: %s, Result: %s", doi, message_body)
    doi_process_attempt.sqs_error_update_status()  # new method on DoiProcessAttempt
else:
    logger.info("DOI: %s, Result: %s", doi, message_body)
    doi_process_attempt.sqs_success_update_status() # new method on DoiProcessAttempt
```